service/awsconfig/v3/resource_set.go

```diff
@@ -0,0 +1,309 @@
+package v3
+
+import (
+	"github.com/cenkalti/backoff"
+	"github.com/giantswarm/certs/legacy"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/giantswarm/operatorkit/framework"
+	"github.com/giantswarm/operatorkit/framework/resource/metricsresource"
+	"github.com/giantswarm/operatorkit/framework/resource/retryresource"
+	"github.com/giantswarm/randomkeytpr"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/giantswarm/aws-operator/client/aws"
+	awsservice "github.com/giantswarm/aws-operator/service/aws"
+	"github.com/giantswarm/aws-operator/service/awsconfig/v2/key"
+	"github.com/giantswarm/aws-operator/service/awsconfig/v2/resource/cloudformation"
+	"github.com/giantswarm/aws-operator/service/awsconfig/v2/resource/cloudformation/adapter"
+	"github.com/giantswarm/aws-operator/service/awsconfig/v2/resource/endpoints"
+	"github.com/giantswarm/aws-operator/service/awsconfig/v2/resource/kmskey"
+	"github.com/giantswarm/aws-operator/service/awsconfig/v2/resource/namespace"
+	"github.com/giantswarm/aws-operator/service/awsconfig/v2/resource/s3bucket"
+	"github.com/giantswarm/aws-operator/service/awsconfig/v2/resource/service"
+	"github.com/giantswarm/aws-operator/service/awsconfig/v3/resource/s3object"
+	// TODO v2 cloudconfig is used in legacy resource, v3 cloudconfig is used in v2 resources and v4 cloudconfig in v3 resources.
+	"github.com/giantswarm/aws-operator/service/awsconfig/v4/cloudconfig"
+)
+
+const (
+	ResourceRetries uint64 = 3
+)
+
+type ResourceSetConfig struct {
+	CertsSearcher      legacy.Searcher
+	GuestAWSClients    aws.Clients
+	HostAWSClients     aws.Clients
+	K8sClient          kubernetes.Interface
+	Logger             micrologger.Logger
+	RandomkeysSearcher *randomkeytpr.Service
+
+	HandledVersionBundles []string
+	InstallationName      string
+	ProjectName           string
+}
+
+func NewResourceSet(config ResourceSetConfig) (*framework.ResourceSet, error) {
+	var err error
+
+	if config.CertsSearcher == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.CertsSearcher must not be empty")
+	}
+	if config.GuestAWSClients.CloudFormation == nil {
```

Comment:
Totally unrelated to this PR but just to have this said. In the future it would probably make sense to have a `Validate` method for such things as soon as we wrap stuff like we do here. The reason for doing this can be seen below.

Reply:
Good point. Is there an example of us using `Validate` or is this a new approach?

Reply:
The flag structures in opsctl do it that way: https://github.com/giantswarm/opsctl/blob/master/command/create/kubeconfig/flags.go.

Reply:
azure-operator client package uses this approach: https://github.com/giantswarm/azure-operator/blob/a6cad0dccf66faf7260f4b28ca87acb058eb79cc/client/azure.go#L48

Then example usage: https://github.com/giantswarm/azure-operator/blob/a6cad0dccf66faf7260f4b28ca87acb058eb79cc/service/operator/service.go#L60

Do we want to do it like that? Note there is not masking in `Validate`, because it doesn't make sense. Should we do something like that to all configs?

Reply:
Having `Validate` would be useful here because there is a lot of config to pass in. I don't think it's needed / worth changing it everywhere. But I don't have a strong opinion on this.

Reply:
Yeah, I rather ask if that should become our goto apprach for every new Config struct.